insights-notification-schemas-java/src/test/java/com/redhat/cloud/notifications/ingress/TestSerialization.java

```diff
@@ -132,13 +133,30 @@ void shouldHaveDefaultValuesWhenNotSet() {
+         Action action = Parser.decode(jsonAction);
+         assertTrue(action.getEvents().get(0).getAdditionalProperties().containsKey("report_url"));
+
+         assertThrows(ParsingException.class, () -> Parser.encode(action));
+         assertThrows(ParsingException.class, () -> Parser.validate(action));
 
-         assertDoesNotThrow(() -> Parser.decode(Parser.encode(action)));
+         // Should throw ParsingException but does throw ClassCastException
+         // fixed by: https://github.com/networknt/json-schema-validator/pull/561
+         assertThrows(Exception.class, () -> Parser.validate(jsonAction));
      }
 
+    @Test
+    void shouldAcceptTimestampWith00OffsetWhenDeserializingAndFailIfValidating() {
+        String jsonAction = "{\"version\":\"v1.1.0\",\"bundle\":\"rhel\",\"application\":\"compliance\",\"event_type\":\"compliance-below-threshold\",\"timestamp\":\"2022-05-02T17:30:54+00:00\",\"account_id\":\"6089719\",\"events\":[{\"metadata\":{},\"payload\":\"{\\\"host_id\\\":\\\"597f855d-2abc-412c-aabc-8213e0f52655\\\",\\\"host_name\\\":\\\"rhiqe.g1myw1.test\\\",\\\"policy_id\\\":\\\"7c247099-4e1b-42e6-adaa-022e026bd6c3\\\",\\\"policy_name\\\":\\\"Red Hat Corporate Profile for Certified Cloud Providers (RH CCP)\\\",\\\"policy_threshold\\\":100.0,\\\"compliance_score\\\":50.5}\"}],\"context\":\"{\\\"display_name\\\":\\\"rhiqe.g1myw1.test\\\",\\\"host_url\\\":\\\"https://console.redhat.com/insights/inventory/597f855d-2abc-412c-aabc-8213e0f52655\\\",\\\"inventory_id\\\":\\\"597f855d-2abc-412c-aabc-8213e0f52655\\\",\\\"rhel_version\\\":\\\"7.8\\\",\\\"tags\\\":[{\\\"key\\\":\\\"pnzXd\\\",\\\"value\\\":\\\"fjMNEWZq\\\",\\\"namespace\\\":\\\"foRxKl\\\"},{\\\"key\\\":\\\"QTCudz\\\",\\\"value\\\":\\\"PmUkZUgAW\\\",\\\"namespace\\\":\\\"wQAFzbA\\\"},{\\\"key\\\":\\\"KivqqjEz\\\",\\\"value\\\":\\\"sRLKnWkj\\\",\\\"namespace\\\":\\\"AxWVzGyCK\\\"},{\\\"key\\\":\\\"pnYHATPvHo\\\",\\\"value\\\":\\\"nRBuWbU\\\",\\\"namespace\\\":\\\"XnvpVIWHZ\\\"},{\\\"key\\\":\\\"tQfAQQ\\\",\\\"value\\\":\\\"agDLlxVKl\\\",\\\"namespace\\\":\\\"KvHmTtqgzh\\\"}]}\",\"recipients\":[]}";
+        Action action = Parser.decode(jsonAction);
+        assertEquals(LocalDateTime.of(2022, 5, 2, 17, 30 ,54), action.getTimestamp());
+
+        // Should throw ParsingException but does throw ClassCastException
+        // fixed by: https://github.com/networknt/json-schema-validator/pull/561
+        assertThrows(Exception.class, () -> Parser.validate(jsonAction));
```

Comment:
Same question.

Reply:
Updated - the ClassCastException wasn't relevant in this context. It was throw because we were sending payload and context as strings - these are only supported on the decode.